internal/agent/device/lifecycle/manager.go

```diff
@@ -92,15 +96,71 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warnf("Detected decommissioning request from flightctl service")
+		errored := false
+		m.updateWithStartedCondition(ctx)
+		m.log.Warnf("Updated Condition to decommissioning started")
+
+		// todo - switch handling based on decommission target type once we support more types;
+		// keep track of whether we have an error
+
+		if !errored {
+			m.log.Warnf("No errors during decommissioning, updating Condition to decommissioning completed")
+			m.updateWithCompletedCondition(ctx)
+		}
+
+		// no return
+		m.log.Warnf("Preparing to wipe agent certificate and keys and reboot")
+		m.wipeAndReboot()
+	}
+
 	return nil
 }
 
+func (m *LifecycleManager) updateWithStartedCondition(ctx context.Context) {
+	updateErr := m.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
+		Type:    v1alpha1.DeviceDecommissioning,
+		Status:  v1alpha1.ConditionStatusTrue,
+		Reason:  string(v1alpha1.DecommissionStateStarted),
+		Message: "The device has started decommissioning",
+	})
+	if updateErr != nil {
+		m.log.Warnf("Failed setting status: %v", updateErr)
+	}
+}
+
+func (m *LifecycleManager) updateWithCompletedCondition(ctx context.Context) {
+	updateErr := m.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
+		Type:    v1alpha1.DeviceDecommissioning,
+		Status:  v1alpha1.ConditionStatusTrue,
+		Reason:  string(v1alpha1.DecommissionStateComplete),
+		Message: "The device has completed decommissioning and will wipe its management certificate",
+	})
+	if updateErr != nil {
+		m.log.Warnf("Failed setting status: %v", updateErr)
+	}
+}
+
+// point of no return - wipes management cert and keys
+func (m *LifecycleManager) wipeAndReboot() {
+	err := m.deviceReadWriter.RemoveFile(m.managementCertPath)
+	if err != nil {
+		m.log.Errorf("Unable to delete mcert: %v", err)
+	}
+
+	m.enrollmentUIEndpoint = ""
+	m.enrollmentClient = nil
+	m.enrollmentCSR = nil
+
+	// todo - reboot
```

Comment:
I think this is critical right? To do this you will need an executor to send a command via os exec. Or extend the systemd client.

https://github.com/flightctl/flightctl/blob/main/internal/agent/client/systemd.go#L18